addons/mail/static/src/web/chatter.xml

```diff
@@ -153,7 +153,7 @@
 </t>
 
 <t t-name="mail.Chatter.attachFiles" owl="1">
-    <button class="o-mail-Chatter-attachFiles btn btn-link text-action px-1 d-flex align-items-center" aria-label="Attach files" t-att-class="{ 'my-2': !props.compactHeight }" t-on-click="onClickAddAttachments">
+    <button class="o-mail-Chatter-attachFiles btn btn-link text-action px-1 d-flex align-items-center" aria-label="Attach files" t-att-class="{ 'my-2': !props.compactHeight }" t-att-disabled="!state.thread.hasWriteAccess and attachments.length === 0" t-on-click="onClickAddAttachments">
```

Comment:
If no access rights and there are some attachments, then we can upload more attachments?
Only `hasWriteAccess` rights should be used here. It's okay to show disabled when `hasWriteAccess` is `undefined`, optimistic value (= value when not yet fetched) should certainly be disabled, to prevent uploading if in the end the user cannot upload

Reply:
No, if there are no access rights then you cant upload more attachments.

The reason for `attachments.length === 0` is that if there are some attachments there then clicking on this button will open the `AttachmentBox`.